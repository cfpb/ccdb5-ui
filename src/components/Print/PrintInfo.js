import './PrintInfo.scss';
import { useSelector } from 'react-redux';
import { useMemo } from 'react';
import { shortFormat } from '../../utils';
import {
  selectQueryDateReceivedMax,
  selectQueryDateReceivedMin,
  selectQuerySearchText,
} from '../../reducers/query/selectors';
import { selectViewIsPrintMode } from '../../reducers/view/selectors';
import { useGetAggregations } from '../../api/hooks/useGetAggregations';

export const PrintInfo = () => {
  const { data } = useGetAggregations();
  const dateMin = useSelector(selectQueryDateReceivedMin);
  const dateMax = useSelector(selectQueryDateReceivedMax);
  const dateText = shortFormat(dateMin) + ' - ' + shortFormat(dateMax);

  const searchText = useSelector(selectQuerySearchText);
  const isPrintMode = useSelector(selectViewIsPrintMode);
  const docCount = data?.doc_count || 0;
  const total = data?.total || 0;

  const complaintCountText = useMemo(() => {
    if (docCount === total) {
      return (
        <div>
          Showing <span>{total.toLocaleString()}</span> complaints
        </div>
      );
    }
    return (
      <div>
        Showing <span>{total.toLocaleString()}</span> out of
        <span> {docCount.toLocaleString()} </span> total complaints{' '}
      </div>
    );
  }, [docCount, total]);

  return isPrintMode ? (
    <section className="print-info">
      <p>
        <span>Dates:</span> {dateText}
      </p>
      {!!searchText && (
        <p>
          <span>Search Term:</span> {searchText}
        </p>
      )}
      <div>{complaintCountText}</div>
    </section>
  ) : null;
};
