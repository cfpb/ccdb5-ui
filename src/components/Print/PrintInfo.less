@import (less) '../../css/base.less';

.print-info, .print-info-footer {
  width: 100%;
  background: @gray-5;
  word-break: break-all;
  padding: 15px;
  span {
    font-weight: 600;
  }
}
