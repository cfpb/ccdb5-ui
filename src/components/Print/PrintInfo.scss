.print-info,
.print-info-footer {
  width: 100%;
  background: var(--gray-5);
  word-break: break-all;
  padding: 15px;
  span {
    font-weight: 600;
  }
}
