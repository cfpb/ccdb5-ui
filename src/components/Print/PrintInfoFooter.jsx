import { selectViewIsPrintMode } from '../../reducers/view/selectors';
import { useSelector } from 'react-redux';

export const PrintInfoFooter = () => {
  const isPrintMode = useSelector(selectViewIsPrintMode);
  return isPrintMode ? (
    <section className="print-info-footer">
      <p>
        <span>URL:</span> {window.location.href}
      </p>
    </section>
  ) : null;
};
