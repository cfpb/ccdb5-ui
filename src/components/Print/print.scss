.print {
  .content__main {
    display: block;
    width: 100% !important;
  }

  .o-header {
    min-height: initial;
  }

  .m-global-header-cta,
  .content__wrapper {
    border: none !important;
  }

  .pill-panel li {
    margin: 3px;
  }

  .m-global-eyebrow,
  .m-global-header-cta,
  .action-bar,
  .search-panel h2,
  .pill svg,
  .clear-all,
  .search-bar,
  .content__hero,
  .content__sidebar,
  .tabbed-navigation,
  .filter-panel hr,
  .refine-bar,
  .single-checkbox,
  .state-heading .clear,
  .state-navigation,
  .m-pagination,
  .u-right,
  .total,
  #clear-focus,
  .trend-depth-toggle {
    display: none !important;
  }

  .advanced-container,
  .search-panel {
    width: 100%;
  }

  #tile-chart-map {
    height: 700px;
    width: 700px;
  }
  #line-chart,
  #stacked-area-chart {
    width: 550px;
  }

  .tooltip-container {
    margin-top: 50px;
    width: 100px;
    .tooltip-ul {
      span.u-left {
        width: 100% !important;
        text-decoration: none !important;
      }
    }
  }

  .map-toolbar,
  .row-chart-section {
    page-break-before: always;
  }
}

@media print {
  .m-global-eyebrow,
  .m-global-header-cta,
  .action-bar,
  .search-panel h2,
  .pill svg,
  .clear-all,
  .search-bar,
  .content__hero,
  .content__sidebar,
  .tabbed-navigation,
  .filter-panel hr,
  .refine-bar,
  .single-checkbox,
  .state-heading .clear,
  .state-navigation,
  .m-pagination,
  .warning,
  .u-right,
  .total,
  .vertical-marker,
  .circle-container,
  .a-micro-copy,
  .filter-button,
  footer,
  #clear-focus,
  .filter-toggle-button,
  .trend-depth-toggle {
    display: none !important;
  }
}
