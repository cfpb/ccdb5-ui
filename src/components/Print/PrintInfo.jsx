import './PrintInfo.less';
import { connect } from 'react-redux';
import React from 'react';
import { shortFormat } from '../../utils';

export class PrintInfo extends React.Component {
  render() {
    const { complaintCountText, dates, searchText } = this.props;
    return (
      <section className='print-info'>
        <p><span>Dates:</span> { dates }</p>
        { searchText && <p><span>Search Term:</span> { searchText }</p> }
        <div>{ complaintCountText }</div>
      </section>
    );
  }
}

export const getComplaintCountText = aggs => {
  const { doc_count: docCount, total } = aggs;
  if ( docCount === total ) {
    return (
      <div>Showing <span>{ total.toLocaleString() }</span> complaints</div>
    );
  }
  return (
    <div>Showing <span>{ total.toLocaleString() }</span> out of
      <span> { docCount.toLocaleString() } </span> total complaints </div>
  );
};

const getDateText = query => {
  const { date_received_min: dateMin, date_received_max: dateMax } = query;
  return shortFormat( dateMin ) + ' - ' + shortFormat( dateMax );
};

export const mapStateToProps = state => {
  const { aggs, query } = state;
  return {
    complaintCountText: getComplaintCountText( aggs ),
    dates: getDateText( query ),
    searchText: query.searchText
  };
};

export default connect( mapStateToProps )( PrintInfo );
