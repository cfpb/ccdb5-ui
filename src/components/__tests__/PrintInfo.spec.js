import { getComplaintCountText, mapStateToProps } from '../Print/PrintInfo';
import React from 'react';

describe('component: PrintInfo', () => {
  describe('mapStateToProps', () => {
    it('maps state and props', () => {
      const state = {
        aggs: {
          doc_count: 4000,
          total: 1000,
        },
        query: {
          date_received_max: '2020-03-05T05:00:00.000Z',
          date_received_min: '2017-03-05T05:00:00.000Z',
          searchText: 'foobar',
        },
      };
      const actual = mapStateToProps(state);
      expect(actual.complaintCountText).toBeTruthy();
      expect(actual).toEqual(
        expect.objectContaining({
          dates: '3/5/2017 - 3/5/2020',
          searchText: 'foobar',
        })
      );
    });
  });

  describe('getComplaintCountText', () => {
    it('gets text when counts are filtered', () => {
      const aggs = {
        doc_count: 100,
        total: 100,
      };
      const result = getComplaintCountText(aggs);
      expect(result).toEqual(
        <div>
          Showing <span>100</span> complaints
        </div>
      );
    });
  });
});
