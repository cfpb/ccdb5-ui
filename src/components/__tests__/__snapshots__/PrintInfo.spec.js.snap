// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: PrintInfo initial state renders without crashing 1`] = `
<section
  className="print-info"
>
  <p>
    <span>
      Dates:
    </span>
     
    3/5/2017 - 3/5/2020
  </p>
  <p>
    <span>
      Search Term:
    </span>
     
    Got no money
  </p>
  <div>
    <div>
      Showing 
      <span>
        100
      </span>
       out of
      <span>
         
        1,000
         
      </span>
       total complaints
       
    </div>
  </div>
</section>
`;
